### PR TITLE
feat: Added byteswap to numpy frontend

### LINKED ISSUE #23782: byteswap

### PR DESCRIPTION
**Pull request by VedangSavadi**
### Related issue
Close #23782

### Description

This pull request introduces a  numpy instance method, byteswap, to the ndarray class in the ivy.functional.frontends.numpy.ndarray module. The byteswap method provides the capability to swap the byte order of the data in the ndarray.

Details:

Method Signature:

byteswap(self, inplace=False)
Parameters:

self: The ndarray instance on which this method is called.
inplace (optional, default False): A boolean flag that determines whether the byteswap should be performed in-place or return a new ndarray with swapped bytes.
Functionality:

When inplace is True, the method performs an in-place byteswap operation on the ndarray object.
When inplace is False, the method returns a new ndarray with the byte order swapped.

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by VedangSavadi** [comment]
Hi @RickSanchezStoic , waiting for review and feedback. Thank you!

**Comment by RickSanchezStoic** [comment]
@RashulChutani why it `dill` not being found in the tests even when it's a requirement listed ?

**Comment by VedangSavadi** [comment]
Hi, @RickSanchezStoic please let me know if any changes are required. Thank you!

**Comment by RickSanchezStoic** [comment]
@RashulChutani the test function `test_numpy_ndarray_byteswap` isn't getting detected it seems, can you please look into this once? Thanks!

**Comment by Ishticode** [comment]
pls resolve the conflict and i will take a look at the ci and merge it asap as its a pretty straightforward one

**Comment by VedangSavadi** [comment]
@Ishticode I have resolved the conflicts now. Thank You

File: ivy/functional/frontends/numpy/ndarray/ndarray.py [hunk 1]
```diff
@@ -402,6 +402,13 @@ def sum(
             out=out,
         )
 
+    def byteswap(self, inplace=False):
+        if inplace:
+            self.ivy_array.byteswap(True)
+            return self
+        else:
+            return self.ivy_array.byteswap()
```
Comment (Sam-Armstrong) [inline ivy/functional/frontends/numpy/ndarray/ndarray.py:410]: There is no `byteswap` method in the ivy api right now, so this frontend function would require a manual implementation using other ivy functions.

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]

**Pull request closed without merging**